qemu/tests/hugepage_mem_stress.py

```diff
@@ -29,40 +35,55 @@ def heavyload_install():
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+
     os_type = params["os_type"]
     verify_wait_timeout = params.get_numeric("verify_wait_timeout", 60)
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
     session = vm.wait_for_login()
 
-    error_context.context("Run memory heavy stress in guest", logging.info)
-    if os_type == "linux":
-        stress_args = params["stress_custom_args"] % (
-                params.get_numeric("mem") / 512)
-        stress_test = utils_test.VMStress(vm, "stress",
-                                          params, stress_args=stress_args)
-        try:
-            stress_test.load_stress_tool()
-            utils_misc.wait_for(lambda: (stress_test.app_running is False), 30)
-            stress_test.unload_stress()
-            utils_misc.verify_dmesg(session=session)
-        finally:
-            stress_test.clean()
-    else:
-        install_path = params["install_path"]
-        test_installed_cmd = 'dir "%s" | findstr /I heavyload' % install_path
-        heavyload_install()
-        error_context.context("Run heavyload inside guest.", logging.info)
-        heavyload_bin = r'"%s\heavyload.exe" ' % install_path
-        heavyload_options = ["/MEMORY %d" % (params.get_numeric("mem") / 512),
-                             "/DURATION 30",
-                             "/AUTOEXIT",
-                             "/START"]
-        start_cmd = heavyload_bin + " ".join(heavyload_options)
-        stress_tool = BackgroundTest(session.cmd, (start_cmd, 30, 30))
-        stress_tool.start()
-        if not utils_misc.wait_for(stress_tool.is_alive, verify_wait_timeout):
-            test.error("Failed to start heavyload process.")
-        stress_tool.join(30)
+    try:
+        error_context.context("Run memory heavy stress in guest", logging.info)
+        if os_type == "linux":
+            stress_args = params["stress_custom_args"] % (
+                    params.get_numeric("mem") / 512)
+            stress_test = utils_test.VMStress(vm, "stress",
+                                              params, stress_args=stress_args)
+            try:
+                stress_test.load_stress_tool()
+                utils_misc.wait_for(lambda: (stress_test.app_running is False), 30)
+                stress_test.unload_stress()
+                utils_misc.verify_dmesg(session=session)
+            finally:
+                stress_test.clean()
+        else:
+            install_path = params["install_path"]
+            test_installed_cmd = 'dir "%s" | findstr /I heavyload' % install_path
+            heavyload_install()
+            error_context.context("Run heavyload inside guest.", logging.info)
+            heavyload_bin = r'"%s\heavyload.exe" ' % install_path
+            heavyload_options = ["/MEMORY %d" % (params.get_numeric("mem") / 512),
+                                 "/DURATION 30s",
```

Comment:
I tried and the minimum limit is `1`, it required a interger, so you need to modify it.

Reply:
ok, thanks for the reminder.